config/3.3/aiManager/templates/gobal-image-pull-secret.yaml

```diff
@@ -0,0 +1,125 @@
+              cpu: "500m"
+          command:
+            - /bin/sh
+            - -c
+            - |
+              set -eo pipefail
+              set -x
+
+              ARTIFACTORY_EMAIL={{.Values.ARTIFACTORY_EMAIL}}
+              ARTIFACTORY_KEY={{.Values.ARTIFACTORY_KEY}}
+              DOCKER_USER={{.Values.DOCKER_USER}}
+              DOCKER_KEY={{.Values.DOCKER_KEY}}
+              CP_STG_USER={{.Values.CP_STG_USER}}
+              CP_PROD_USER={{.Values.CP_PROD_USER}}
+              CP_PROD_KEY={{.Values.CP_PROD_KEY}}
+              CP_STG_KEY={{.Values.CP_STG_KEY}}
+
+              if [[ "$ARTIFACTORY_EMAIL" == "" || "$ARTIFACTORY_KEY" == "" || "$CP_PROD_KEY" == "" || "$CP_STG_KEY" == "" ]]; then
+                  echo "Need image pull secret for internal build: ARTIFACTORY_EMAIL,ARTIFACTORY_KEY,CP_PROD_KEY,CP_STG_KEY"
+                  exit 1
+              fi
+
+              ARTIFACTORY_AUTH="$ARTIFACTORY_EMAIL:$ARTIFACTORY_KEY"
+              DOCKER_AUTH="$DOCKER_USER:$DOCKER_KEY"
+              CP_STG_USER="${CP_STG_USER:=iamapikey}"
+              CP_PROD_USER="${CP_PROD_USER:=iamapikey}"
+              CP_PROD_AUTH="$CP_PROD_USER:$CP_PROD_KEY"
+              CP_STG_AUTH="$CP_STG_USER:$CP_STG_KEY"
+
+              # Extract the existing pull secret from cluster
+              rm -rf ${HOME}/.dockerconfigjson
+              oc extract secret/pull-secret -n openshift-config --to ${HOME} --confirm
+              if [[ $? -ne 0 ]]; then
+                  logger "[ERROR] Failed to extract existing pull-secret!"
+                  return 1
+              fi
+
+              # Patch .dockerconfigjson with user values
+              oc registry login --registry hyc-katamari-cicd-team-docker-local.artifactory.swg-devops.com \
+              --auth-basic=$ARTIFACTORY_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              oc registry login --registry hyc-cloud-private-daily-docker-local.artifactory.swg-devops.com \
+              --auth-basic=$ARTIFACTORY_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              oc registry login --registry hyc-hdm-devops-team-docker-local.artifactory.swg-devops.com \
+              --auth-basic=$ARTIFACTORY_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              oc registry login --registry docker.io \
+              --auth-basic=$DOCKER_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              oc registry login --registry cp.icr.io \
+              --auth-basic=$CP_PROD_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              oc registry login --registry cp.stg.icr.io \
+              --auth-basic=$CP_STG_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              # Apply the new secret
+              oc set data secret/pull-secret --from-file .dockerconfigjson=${HOME}/.dockerconfigjson -n openshift-config
+
+              wait_time=40
+              # master nodes
+              sleep $wait_time
```

Comment:
Just to confirm, in which version do we need to wait for node restart, as for OCP 4.8.27, we've seen this is not needed any more.

Reply:
Probably required for versions before 4.7, not very sure. But we want to make a universally applicable template, I think add a judgment will be more comprehensive.